### PR TITLE
Rst view bob

### PR DESCRIPTION
**Pull request by ryansteeters**
Altered set_matrix_3d to accept isWalking, so that viewBobbing can be applied only when the user is in motion.

### CODE REVIEW

**Review by Dustin-Layne — COMMENTED** [review]
Comments seem to explain each method and formatting looks good. I see no obvious errors within the code I'm reading. It's nice to see the isWalking boolean I added being using in other features so easily, nice work.

File: src/main.c [hunk 1]
```diff
@@ -1338,7 +1342,7 @@ void ensure_chunks_worker(Player *player, Worker *worker) {
     float matrix[16];
     set_matrix_3d(
         matrix, g->width, g->height,
-        s->x, s->y, s->z, s->rx, s->ry, g->fov, g->ortho, g->render_radius);
+        s->x, s->y, s->z, s->rx, s->ry, g->fov, g->ortho, g->render_radius, g->isBobbing);
```
Comment (jrichards1999) [inline src/main.c:1345]: Nice work taking advantage of existing code to add new features.

File: src/matrix.c [hunk 2]
```diff
@@ -210,23 +214,46 @@ void set_matrix_2d(float *matrix, int width, int height) {
...
 
-float viewBob_offSet(float y) {
-	matrix_viewHeight = y;	    
-	return matrix_viewHeight;
+
+
+///
+///viewBob_offSet returns a modified value for the camera to offset while the player is walking
+///
+float viewBob_offSet(float y,_Bool isWalking) {
```
Comment (jrichards1999) [inline src/matrix.c:222]: Doxygen comments are helping a lot with code readability!

**Review by jrichards1999 — COMMENTED** [review]
The code looks great, Ryan! Lots of code to take in but it's well commented and understandable.  It looks like you have some print to the terminal that I assume are for test code coverage?  I suggest making them a bit more descriptive.  For example instead of printing either 'true' or 'false' for isWalking, try printing out 'isWalking = true'

**Comment by ryansteeters** [comment]
Thanks for the comments, guys! I'm going go ahead an merge the request

**Comment by mistcurve** [comment]
Good job! it good job commenting and making things readable. I like how you changed the viewbobbing function to make it more logical.

**Pull request merged**